<!-- 
RSS generated by JIRA (6.3.4#6332-sha1:51bc225ef474afe3128b2f66878477f322397b16) at Sun May 17 03:43:14 UTC 2015

It is possible to restrict the fields that are returned in this document by specifying the 'field' parameter in your request.
For example, to request only the issue key and summary add field=key&field=summary to the URL of your request.
For example:
https://issues.apache.org/jira/si/jira.issueviews:issue-xml/DERBY-2854/DERBY-2854.xml?field=key&amp;field=summary
-->
<rss version="0.92" >
<channel>
    <title>ASF JIRA</title>
    <link>https://issues.apache.org/jira</link>
    <description>This file is an XML representation of an issue</description>
    <language>en-uk</language>    <build-info>
        <version>6.3.4</version>
        <build-number>6332</build-number>
        <build-date>15-08-2014</build-date>
    </build-info>

<item>
            <title>[DERBY-2854] Convert jdbc4/TestResultSetMethods.java to JUnit</title>
                <link>https://issues.apache.org/jira/browse/DERBY-2854</link>
                <project id="10594" key="DERBY">Derby</project>
                    <description></description>
                <environment></environment>
        <key id="12372152">DERBY-2854</key>
            <summary>Convert jdbc4/TestResultSetMethods.java to JUnit</summary>
                <type id="3" iconUrl="https://issues.apache.org/jira/images/icons/issuetypes/task.png">Task</type>
                                            <priority id="3" iconUrl="https://issues.apache.org/jira/images/icons/priorities/major.png">Major</priority>
                        <status id="6" iconUrl="https://issues.apache.org/jira/images/icons/statuses/closed.png" description="The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.">Closed</status>
                    <statusCategory id="3" key="done" colorName="green"/>
                                    <resolution id="1">Fixed</resolution>
                                        <assignee username="moazeni">Ramin Moazeni</assignee>
                                    <reporter username="moazeni">Ramin Moazeni</reporter>
                        <labels>
                    </labels>
                <created>Thu, 21 Jun 2007 18:33:55 +0100</created>
                <updated>Fri, 21 Jan 2011 17:50:26 +0000</updated>
                            <resolved>Wed, 16 Jan 2008 17:53:44 +0000</resolved>
                                    <version>10.3.1.4</version>
                                    <fixVersion>10.4.1.3</fixVersion>
                                    <component>Test</component>
                        <due></due>
                            <votes>0</votes>
                                    <watches>0</watches>
                                                                <comments>
                            <comment id="12509920" author="moazeni" created="Tue, 3 Jul 2007 16:58:44 +0100"  >&lt;p&gt;Attachment please find the patch that converts this issue to JUnit.&lt;br/&gt;
You feedback and review is appreciated.&lt;/p&gt;

&lt;p&gt;Please note that derbyall and suites.All tests are finished without&lt;br/&gt;
any errors.&lt;/p&gt;</comment>
                            <comment id="12510147" author="knutanders" created="Wed, 4 Jul 2007 12:26:44 +0100"  >&lt;p&gt;I&apos;m not able to build Derby with the patch.&lt;/p&gt;

&lt;p&gt;compile_jdbc4:&lt;br/&gt;
    &lt;span class=&quot;error&quot;&gt;&amp;#91;javac&amp;#93;&lt;/span&gt; Compiling 28 source files to /export/home/kh160127/derby/trunk/classes&lt;br/&gt;
    &lt;span class=&quot;error&quot;&gt;&amp;#91;javac&amp;#93;&lt;/span&gt; /export/home/kh160127/derby/trunk/java/testing/org/apache/derbyTesting/functionTests/tests/jdbc4/_Suite.java:70: cannot find symbol&lt;br/&gt;
    &lt;span class=&quot;error&quot;&gt;&amp;#91;javac&amp;#93;&lt;/span&gt; symbol  : variable TestResultSetMethods&lt;br/&gt;
    &lt;span class=&quot;error&quot;&gt;&amp;#91;javac&amp;#93;&lt;/span&gt; location: class org.apache.derbyTesting.functionTests.tests.jdbc4._Suite&lt;br/&gt;
    &lt;span class=&quot;error&quot;&gt;&amp;#91;javac&amp;#93;&lt;/span&gt;             suite.addTest(TestResultSetMethods.suite());&lt;br/&gt;
    &lt;span class=&quot;error&quot;&gt;&amp;#91;javac&amp;#93;&lt;/span&gt;                           ^&lt;br/&gt;
    &lt;span class=&quot;error&quot;&gt;&amp;#91;javac&amp;#93;&lt;/span&gt; Note: Some input files use unchecked or unsafe operations.&lt;br/&gt;
    &lt;span class=&quot;error&quot;&gt;&amp;#91;javac&amp;#93;&lt;/span&gt; Note: Recompile with -Xlint:unchecked for details.&lt;br/&gt;
    &lt;span class=&quot;error&quot;&gt;&amp;#91;javac&amp;#93;&lt;/span&gt; 1 error&lt;/p&gt;

&lt;p&gt;BUILD FAILED&lt;/p&gt;</comment>
                            <comment id="12510155" author="knutanders" created="Wed, 4 Jul 2007 13:00:40 +0100"  >&lt;p&gt;Some general comments:&lt;/p&gt;

&lt;p&gt;  1) The argument order is wrong for the calls to assertEquals(). The expected value should come before the actual value.&lt;br/&gt;
  2) The baseSuite() method could be removed, and suite() could be implemented as a simple call to TestConfiguration.defaultSuite(ResultSetMethodsTest.class).&lt;br/&gt;
  3) There&apos;s no need for the savedAutoCommit field since each fixture gets a new connection.&lt;br/&gt;
  4) Catching exceptions just to call fail() should be avoided as it will swallow the stack trace. The exception should rather be propagated up the call stack and handled by the JUnit framework.&lt;br/&gt;
  5) assertEquals(...true...) and assertEquals(...false...) should be replaced with assertTrue() and assertFalse().&lt;br/&gt;
  6) When assertSQLState() is called, there is no need to call assertNotNull(sqle.getSQLState()).&lt;/p&gt;</comment>
                            <comment id="12510158" author="oysteing" created="Wed, 4 Jul 2007 13:15:07 +0100"  >&lt;p&gt;Note that there is also an existing junit test jdbc4/ResultSetTest.&lt;br/&gt;
I think it is a bit confusing to have two different tests with so similar names.&lt;br/&gt;
How should I know which one to use when adding another test case?&lt;br/&gt;
Has it been checked whether these two tests have overlapping test cases?&lt;/p&gt;</comment>
                            <comment id="12510742" author="moazeni" created="Fri, 6 Jul 2007 18:10:03 +0100"  >&lt;p&gt;Thanks Knut for your review. I am hoping that I have addressed&lt;br/&gt;
all your comments in this new patch.&lt;/p&gt;

&lt;p&gt;Your review and comments are appreciated.&lt;/p&gt;</comment>
                            <comment id="12510745" author="moazeni" created="Fri, 6 Jul 2007 18:13:38 +0100"  >&lt;p&gt;Hi Oystein,&lt;/p&gt;

&lt;p&gt;I looked at jdbc4/ResultSetTest and it looks like a different test.&lt;br/&gt;
How about changing jdbc4/ResultSetMethodsTest.java to &lt;br/&gt;
jdbc4/HoldabilityTest.java?&lt;/p&gt;

&lt;p&gt;Thanks&lt;br/&gt;
Ramin&lt;/p&gt;</comment>
                            <comment id="12510748" author="djd" created="Fri, 6 Jul 2007 18:39:00 +0100"  >&lt;p&gt;The test could use the utility methods in most test fixtures to create statement objects.&lt;/p&gt;

&lt;p&gt;Replace&lt;/p&gt;

&lt;p&gt; Statement s = conn.createStatement();&lt;/p&gt;

&lt;p&gt;with&lt;/p&gt;

&lt;p&gt;  Statement s = createStatement();&lt;/p&gt;

&lt;p&gt;e.g. in testGetHoldability().&lt;/p&gt;

&lt;p&gt;These objects will very soon be automatically closed at tearDown and make the test more readable.&lt;/p&gt;</comment>
                            <comment id="12511082" author="oysteing" created="Mon, 9 Jul 2007 12:05:18 +0100"  >&lt;p&gt;I am not sure what you mean by &quot;looks like a different test&quot;.  Any reason the two tests can not be merged?&lt;/p&gt;

&lt;p&gt;Renaming jdbc4/ResultSetMethodsTest.java to jdbc4/HoldabilityTest.java does not seem like a good idea since the holdability test case is just one of several test cases in this test.&lt;/p&gt;</comment>
                            <comment id="12514655" author="moazeni" created="Mon, 23 Jul 2007 16:28:33 +0100"  >&lt;p&gt;Thanks Dan for your review...I am attaching another patch which&lt;br/&gt;
I think addresses your comments. Please review....&lt;/p&gt;

&lt;p&gt;I looked at both ResultSetTest.java and ResultSetMethodsTest.java&lt;br/&gt;
and they both seems to test JDBC4 methods in ResultsSet interface.&lt;br/&gt;
So, I merged the two tests into ResultsSetTest.java.&lt;/p&gt;

&lt;p&gt;Please note that I added some utility methods to BaseJDBCTestCase.java:&lt;br/&gt;
public Statement createStatement(int resultSetType,int resultSetConcurrency,int resultSetHoldability)&lt;br/&gt;
public CallableStatement prepareCall(String sql, int resultSetType,int resultSetConcurrency,int resultSetHoldability)&lt;br/&gt;
public CallableStatement prepareCall(String sql,int resultSetType,int resultSetConcurrency) &lt;/p&gt;

&lt;p&gt;Thanks&lt;br/&gt;
Ramin&lt;/p&gt;</comment>
                            <comment id="12515848" author="kmarsden" created="Thu, 26 Jul 2007 21:14:59 +0100"  >&lt;p&gt;I have not looked at the code yet, but trying to run the test I get:&lt;br/&gt;
&lt;span class=&quot;error&quot;&gt;&amp;#91;C:/test&amp;#93;&lt;/span&gt; .tests.jdbc4.ResultSetTest &lt;br/&gt;
.........................................&lt;br/&gt;
.....E.E...................................&lt;br/&gt;
.............E.E..&lt;br/&gt;
Time: 4.703&lt;br/&gt;
There were 4 errors:&lt;br/&gt;
1) testGetHoldability(org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetTest)java.sql.SQLSyntaxErrorException:&lt;br/&gt;
The class &apos;org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetMethodsTest&apos; does not exist or is inaccessible. Th&lt;br/&gt;
is can happen if the class is not public.&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:91)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.Util.newEmbedSQLException(Util.java:88)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.Util.seeNextException(Util.java:224)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:398)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(TransactionResourceImpl.java:346)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedConnection.handleException(EmbedConnection.java:1572)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.ConnectionChild.handleException(ConnectionChild.java:81)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedPreparedStatement.&amp;lt;init&amp;gt;(EmbedPreparedStatement.java:144)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedCallableStatement.&amp;lt;init&amp;gt;(EmbedCallableStatement.java:68)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedCallableStatement20.&amp;lt;init&amp;gt;(EmbedCallableStatement20.java:85)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedCallableStatement30.&amp;lt;init&amp;gt;(EmbedCallableStatement30.java:61)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedCallableStatement40.&amp;lt;init&amp;gt;(EmbedCallableStatement40.java:44)&lt;br/&gt;
        at org.apache.derby.jdbc.Driver40.newEmbedCallableStatement(Driver40.java:124)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedConnection.prepareCall(EmbedConnection.java:1014)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseJDBCTestCase.prepareCall(BaseJDBCTestCase.java:306)&lt;br/&gt;
        at org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetTest.testGetHoldability(ResultSetTest.java:1517)&lt;br/&gt;
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&lt;br/&gt;
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)&lt;br/&gt;
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)&lt;br/&gt;
        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)&lt;br/&gt;
        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)&lt;br/&gt;
        at junit.extensions.TestSetup.run(TestSetup.java:23)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)&lt;br/&gt;
Caused by: java.sql.SQLException: The class &apos;org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetMethodsTest&apos; doe&lt;br/&gt;
s not exist or is inaccessible. This can happen if the class is not public.&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:13&lt;br/&gt;
5)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)&lt;br/&gt;
        ... 40 more&lt;br/&gt;
Caused by: java.sql.SQLException: Java exception: &apos;org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetMethodsTes&lt;br/&gt;
t: java.lang.ClassNotFoundException&apos;.&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:13&lt;br/&gt;
5)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.Util.newEmbedSQLException(Util.java:88)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.Util.javaException(Util.java:245)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:403)&lt;br/&gt;
        ... 38 more&lt;br/&gt;
Caused by: java.lang.ClassNotFoundException: org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetMethodsTest&lt;br/&gt;
        at java.net.URLClassLoader$1.run(URLClassLoader.java:200)&lt;br/&gt;
        at java.security.AccessController.doPrivileged(Native Method)&lt;br/&gt;
        at java.net.URLClassLoader.findClass(URLClassLoader.java:188)&lt;br/&gt;
        at java.lang.ClassLoader.loadClass(ClassLoader.java:306)&lt;br/&gt;
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:276)&lt;br/&gt;
        at java.lang.ClassLoader.loadClass(ClassLoader.java:251)&lt;br/&gt;
        at java.lang.ClassLoader.loadClassInternal(ClassLoader.java:319)&lt;br/&gt;
        at java.lang.Class.forName0(Native Method)&lt;br/&gt;
        at java.lang.Class.forName(Class.java:169)&lt;br/&gt;
        at org.apache.derby.impl.services.reflect.ReflectClassesJava2.loadClassNotInDatabaseJar(ReflectClassesJava2.java&lt;br/&gt;
:108)&lt;br/&gt;
        at org.apache.derby.impl.services.reflect.DatabaseClasses.loadApplicationClass(DatabaseClasses.java:232)&lt;br/&gt;
        at org.apache.derby.iapi.services.loader.ClassInspector.getClass(ClassInspector.java:852)&lt;br/&gt;
        at org.apache.derby.iapi.services.loader.ClassInspector.accessible(ClassInspector.java:130)&lt;br/&gt;
        at org.apache.derby.impl.sql.compile.QueryTreeNode.verifyClassExist(QueryTreeNode.java:1481)&lt;br/&gt;
        at org.apache.derby.impl.sql.compile.StaticMethodCallNode.bindExpression(StaticMethodCallNode.java:248)&lt;br/&gt;
        at org.apache.derby.impl.sql.compile.JavaToSQLValueNode.bindExpression(JavaToSQLValueNode.java:245)&lt;br/&gt;
        at org.apache.derby.impl.sql.compile.CallStatementNode.bindStatement(CallStatementNode.java:145)&lt;br/&gt;
        at org.apache.derby.impl.sql.GenericStatement.prepMinion(GenericStatement.java:314)&lt;br/&gt;
        at org.apache.derby.impl.sql.GenericStatement.prepare(GenericStatement.java:88)&lt;br/&gt;
        at org.apache.derby.impl.sql.conn.GenericLanguageConnectionContext.prepareInternalStatement(GenericLanguageConne&lt;br/&gt;
ctionContext.java:753)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedPreparedStatement.&amp;lt;init&amp;gt;(EmbedPreparedStatement.java:129)&lt;br/&gt;
        ... 33 more&lt;br/&gt;
2) testIsClosed(org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetTest)java.sql.SQLException: PROCEDURE &apos;RETRIE&lt;br/&gt;
VE_RESULT_SETS&apos; already exists.&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:95)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Util.java:202)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:391)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(TransactionResourceImpl.java:346)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedConnection.handleException(EmbedConnection.java:1572)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.ConnectionChild.handleException(ConnectionChild.java:81)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1293)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedStatement.execute(EmbedStatement.java:596)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedStatement.execute(EmbedStatement.java:528)&lt;br/&gt;
        at org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetTest.testIsClosed(ResultSetTest.java:1578)&lt;br/&gt;
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&lt;br/&gt;
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)&lt;br/&gt;
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)&lt;br/&gt;
        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)&lt;br/&gt;
        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)&lt;br/&gt;
        at junit.extensions.TestSetup.run(TestSetup.java:23)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)&lt;br/&gt;
Caused by: java.sql.SQLException: PROCEDURE &apos;RETRIEVE_RESULT_SETS&apos; already exists.&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:13&lt;br/&gt;
5)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)&lt;br/&gt;
        ... 34 more&lt;br/&gt;
Caused by: ERROR X0Y68: PROCEDURE &apos;RETRIEVE_RESULT_SETS&apos; already exists.&lt;br/&gt;
        at org.apache.derby.iapi.error.StandardException.newException(StandardException.java:301)&lt;br/&gt;
        at org.apache.derby.impl.sql.execute.CreateAliasConstantAction.executeConstantAction(CreateAliasConstantAction.j&lt;br/&gt;
ava:237)&lt;br/&gt;
        at org.apache.derby.impl.sql.execute.MiscResultSet.open(MiscResultSet.java:64)&lt;br/&gt;
        at org.apache.derby.impl.sql.GenericPreparedStatement.execute(GenericPreparedStatement.java:370)&lt;br/&gt;
        at org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1203)&lt;br/&gt;
        ... 28 more&lt;br/&gt;
3) testGetHoldability(org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetTest)java.sql.SQLSyntaxErrorException:&lt;br/&gt;
The class &apos;org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetMethodsTest&apos; does not exist or is inaccessible. Th&lt;br/&gt;
is can happen if the class is not public.&lt;br/&gt;
        at org.apache.derby.client.am.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:83)&lt;br/&gt;
        at org.apache.derby.client.am.SqlException.getSQLException(SqlException.java:362)&lt;br/&gt;
        at org.apache.derby.client.am.Connection.prepareCall(Connection.java:1678)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseJDBCTestCase.prepareCall(BaseJDBCTestCase.java:306)&lt;br/&gt;
        at org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetTest.testGetHoldability(ResultSetTest.java:1517)&lt;br/&gt;
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&lt;br/&gt;
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)&lt;br/&gt;
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)&lt;br/&gt;
        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)&lt;br/&gt;
        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)&lt;br/&gt;
        at junit.extensions.TestSetup.run(TestSetup.java:23)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)&lt;br/&gt;
        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)&lt;br/&gt;
        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)&lt;br/&gt;
        at junit.extensions.TestSetup.run(TestSetup.java:23)&lt;br/&gt;
        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)&lt;br/&gt;
        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)&lt;br/&gt;
        at junit.extensions.TestSetup.run(TestSetup.java:23)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)&lt;br/&gt;
Caused by: org.apache.derby.client.am.SqlException: The class &apos;org.apache.derbyTesting.functionTests.tests.jdbc4.ResultS&lt;br/&gt;
etMethodsTest&apos; does not exist or is inaccessible. This can happen if the class is not public.&lt;br/&gt;
        at org.apache.derby.client.am.Statement.completeSqlca(Statement.java:1713)&lt;br/&gt;
        at org.apache.derby.client.net.NetStatementReply.parsePrepareError(NetStatementReply.java:541)&lt;br/&gt;
        at org.apache.derby.client.net.NetStatementReply.parsePRPSQLSTTreply(NetStatementReply.java:143)&lt;br/&gt;
        at org.apache.derby.client.net.NetStatementReply.readPrepare(NetStatementReply.java:76)&lt;br/&gt;
        at org.apache.derby.client.net.StatementReply.readPrepare(StatementReply.java:60)&lt;br/&gt;
        at org.apache.derby.client.net.NetStatement.readPrepare_(NetStatement.java:184)&lt;br/&gt;
        at org.apache.derby.client.am.Statement.readPrepare(Statement.java:1359)&lt;br/&gt;
        at org.apache.derby.client.am.PreparedStatement.readPrepareDescribeInput(PreparedStatement.java:1887)&lt;br/&gt;
        at org.apache.derby.client.am.PreparedStatement.flowPrepareDescribeInputOutput(PreparedStatement.java:1947)&lt;br/&gt;
        at org.apache.derby.client.am.PreparedStatement.prepare(PreparedStatement.java:309)&lt;br/&gt;
        at org.apache.derby.client.am.Connection.prepareCallX(Connection.java:1690)&lt;br/&gt;
        at org.apache.derby.client.am.Connection.prepareCall(Connection.java:1670)&lt;br/&gt;
        ... 36 more&lt;br/&gt;
Caused by: org.apache.derby.client.am.SqlException: Java exception: &apos;org.apache.derbyTesting.functionTests.tests.jdbc4.R&lt;br/&gt;
esultSetMethodsTest: java.lang.ClassNotFoundException&apos;.&lt;br/&gt;
        ... 48 more&lt;br/&gt;
4) testIsClosed(org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetTest)java.sql.SQLException: PROCEDURE &apos;RETRIE&lt;br/&gt;
VE_RESULT_SETS&apos; already exists.&lt;br/&gt;
        at org.apache.derby.client.am.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:95)&lt;br/&gt;
        at org.apache.derby.client.am.SqlException.getSQLException(SqlException.java:362)&lt;br/&gt;
        at org.apache.derby.client.am.Statement.execute(Statement.java:824)&lt;br/&gt;
        at org.apache.derbyTesting.functionTests.tests.jdbc4.ResultSetTest.testIsClosed(ResultSetTest.java:1578)&lt;br/&gt;
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&lt;br/&gt;
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)&lt;br/&gt;
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:95)&lt;br/&gt;
        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)&lt;br/&gt;
        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)&lt;br/&gt;
        at junit.extensions.TestSetup.run(TestSetup.java:23)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)&lt;br/&gt;
        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)&lt;br/&gt;
        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)&lt;br/&gt;
        at junit.extensions.TestSetup.run(TestSetup.java:23)&lt;br/&gt;
        at junit.extensions.TestDecorator.basicRun(TestDecorator.java:22)&lt;br/&gt;
        at junit.extensions.TestSetup$1.protect(TestSetup.java:19)&lt;br/&gt;
        at junit.extensions.TestSetup.run(TestSetup.java:23)&lt;br/&gt;
        at org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)&lt;br/&gt;
Caused by: org.apache.derby.client.am.SqlException: PROCEDURE &apos;RETRIEVE_RESULT_SETS&apos; already exists.&lt;br/&gt;
        at org.apache.derby.client.am.Statement.completeSqlca(Statement.java:1713)&lt;br/&gt;
        at org.apache.derby.client.am.Statement.completeExecuteImmediate(Statement.java:1322)&lt;br/&gt;
        at org.apache.derby.client.net.NetStatementReply.parseEXCSQLIMMreply(NetStatementReply.java:207)&lt;br/&gt;
        at org.apache.derby.client.net.NetStatementReply.readExecuteImmediate(NetStatementReply.java:58)&lt;br/&gt;
        at org.apache.derby.client.net.StatementReply.readExecuteImmediate(StatementReply.java:45)&lt;br/&gt;
        at org.apache.derby.client.net.NetStatement.readExecuteImmediate_(NetStatement.java:125)&lt;br/&gt;
        at org.apache.derby.client.am.Statement.readExecuteImmediate(Statement.java:1318)&lt;br/&gt;
        at org.apache.derby.client.am.Statement.flowExecute(Statement.java:2014)&lt;br/&gt;
        at org.apache.derby.client.am.Statement.executeX(Statement.java:829)&lt;br/&gt;
        at org.apache.derby.client.am.Statement.execute(Statement.java:815)&lt;br/&gt;
        ... 35 more&lt;/p&gt;</comment>
                            <comment id="12516557" author="moazeni" created="Mon, 30 Jul 2007 23:04:59 +0100"  >&lt;p&gt;Thanks Kathey for your comment.&lt;br/&gt;
I am attaching another patch that resolves the problem&lt;br/&gt;
with the earlier patch.&lt;/p&gt;

&lt;p&gt;I appreciate your review.&lt;/p&gt;

&lt;p&gt;Thanks&lt;br/&gt;
Ramin&lt;/p&gt;</comment>
                            <comment id="12519475" author="kmarsden" created="Mon, 13 Aug 2007 18:14:22 +0100"  >&lt;p&gt;Hi Ramin,&lt;/p&gt;

&lt;p&gt;I am having trouble applying the patch, could you synch up, test and recreate the patch?&lt;/p&gt;

&lt;p&gt;Thanks&lt;/p&gt;

&lt;p&gt;Kathey&lt;/p&gt;</comment>
                            <comment id="12520583" author="moazeni" created="Fri, 17 Aug 2007 16:19:25 +0100"  >&lt;p&gt;Thank you Kathey. I recreated a new patch and tested it.&lt;br/&gt;
I appreciate your review.&lt;/p&gt;

&lt;p&gt;Thanks&lt;br/&gt;
Ramin&lt;/p&gt;</comment>
                            <comment id="12559610" author="myrna" created="Wed, 16 Jan 2008 17:53:44 +0000"  >&lt;p&gt;This was committed with revision 574053 - which I assume resolves this issue.&lt;/p&gt;</comment>
                    </comments>
                <issuelinks>
                            <issuelinktype id="12310010">
                    <name>Incorporates</name>
                                                                <inwardlinks description="is part of">
                                        <issuelink>
            <issuekey id="12369744">DERBY-2674</issuekey>
        </issuelink>
                            </inwardlinks>
                                    </issuelinktype>
                    </issuelinks>
                <attachments>
                            <attachment id="12361060" name="DERBY-2854v0.diff" size="36704" author="moazeni" created="Tue, 3 Jul 2007 16:58:44 +0100"/>
                            <attachment id="12361061" name="DERBY-2854v0.stat" size="443" author="moazeni" created="Tue, 3 Jul 2007 16:58:44 +0100"/>
                            <attachment id="12361317" name="DERBY-2854v1.diff" size="37393" author="moazeni" created="Fri, 6 Jul 2007 18:10:03 +0100"/>
                            <attachment id="12361318" name="DERBY-2854v1.stat" size="443" author="moazeni" created="Fri, 6 Jul 2007 18:10:03 +0100"/>
                            <attachment id="12362345" name="DERBY-2854v2.diff" size="38372" author="moazeni" created="Mon, 23 Jul 2007 16:28:33 +0100"/>
                            <attachment id="12362346" name="DERBY-2854v2.stat" size="426" author="moazeni" created="Mon, 23 Jul 2007 16:28:33 +0100"/>
                            <attachment id="12362819" name="DERBY-2854v3.diff" size="38358" author="moazeni" created="Mon, 30 Jul 2007 23:04:58 +0100"/>
                            <attachment id="12362820" name="DERBY-2854v3.stat" size="426" author="moazeni" created="Mon, 30 Jul 2007 23:04:59 +0100"/>
                            <attachment id="12364042" name="DERBY-2854v4.diff" size="38312" author="moazeni" created="Fri, 17 Aug 2007 16:19:25 +0100"/>
                            <attachment id="12364043" name="DERBY-2854v4.stat" size="426" author="moazeni" created="Fri, 17 Aug 2007 16:19:25 +0100"/>
                    </attachments>
                <subtasks>
                    </subtasks>
                <customfields>
                                                <customfield id="customfield_12310310" key="com.atlassian.jira.toolkit:attachments">
                        <customfieldname>Attachment count</customfieldname>
                        <customfieldvalues>
                            <customfieldvalue>10.0</customfieldvalue>
                        </customfieldvalues>
                    </customfield>
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        <customfield id="customfield_12310220" key="com.atlassian.jira.ext.charting:firstresponsedate">
                        <customfieldname>Date of First Response</customfieldname>
                        <customfieldvalues>
                            <customfieldvalue>Wed, 4 Jul 2007 11:26:44 +0000</customfieldvalue>

                        </customfieldvalues>
                    </customfield>
                                                                                                                                                                                                                                                                                                                                                                                    <customfield id="customfield_12310420" key="com.pyxis.greenhopper.jira:gh-global-rank">
                        <customfieldname>Global Rank</customfieldname>
                        <customfieldvalues>
                            <customfieldvalue>30628</customfieldvalue>
                        </customfieldvalues>
                    </customfield>
                                                                                                                                                                                <customfield id="customfield_12311820" key="com.pyxis.greenhopper.jira:gh-lexo-rank">
                        <customfieldname>Rank</customfieldname>
                        <customfieldvalues>
                            <customfieldvalue>2|hy0m8n:</customfieldvalue>

                        </customfieldvalues>
                    </customfield>
                                                                <customfield id="customfield_12310920" key="com.pyxis.greenhopper.jira:gh-global-rank">
                        <customfieldname>Rank (Obsolete)</customfieldname>
                        <customfieldvalues>
                            <customfieldvalue>37421</customfieldvalue>
                        </customfieldvalues>
                    </customfield>
                                                                                                                                                                                                                                        <customfield id="customfield_12310222" key="com.atlassian.jira.ext.charting:timeinstatus">
                        <customfieldname>Time in Status</customfieldname>
                        <customfieldvalues>
                            
                        </customfieldvalues>
                    </customfield>
                                                                                            </customfields>
    </item>
</channel>
</rss>